### PR TITLE
Add memcached recipe

### PR DESCRIPTION
**Pull request by radamanthus**
## Description of your patch

Adds the memcached custom recipe. 

The previous recipe was just a stub. This actually installs memcahced, but it has to be enabled by setting `perform_install` to true in `custom-memcached/attributes/default.rb`.

## Recommended Release Notes

Adds the memcached custom recipe.

## Estimated risk

High. 

Our platform includes memcached in the list of recipes that should be run by default (a legacy from the V4 stack). Breakage on this recipe will cause main chef to fail for all chef runs on V5. To mitigate this, the recipe has an attribute `perform_install` and this is disabled by default.

## Components involved

memcached custom chef recipe

## Description of testing done

See QA instructions

## QA Instructions

Run the test cases in examples/redis/cookbooks/custom-memcached/README.md

### CODE REVIEW

File: cookbooks/memcached/README.md [hunk 1]
```diff
@@ -1,4 +1,11 @@
...
-This is a placeholder stub for memcached that can be overridden.
\ No newline at end of file
+A chef recipe for configuring distributed memcached on [EY Cloud]. This recipe is meant to allow memcached to accept remote connections and add all the servers in your EY Cloud cluster to the `/data/<app_name>/shared/config/memcached.yml` file. 
+
+To make use of this recipe you will also need to add a [deploy hook](https://support.cloud.engineyard.com/entries/21016568-use-deploy-hooks) to your application to create a symlink for the memcached configuration. Just add a line like this to your `before_migrate.rb` hook:
+    
+    run "ln -nfs #{shared_path}/config/memcached_custom.yml #{release_path}/config/memcached.yml"
+    
+
+[EY Cloud]: https://cloud.engineyard.com/extras
```
Comment (dvalfre) [inline cookbooks/memcached/README.md]: This URL gives a 404.

File: examples/memcached/cookbooks/custom-memcached/README.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+## Test Cases
+
+This custom chef recipe has been verified using these test cases:
+
+```
+A. Do not enable the perform_install flag
+  A1. Chef run should not fail
+  A2. memcached should not be running
+B. Install from source
```
Comment (dvalfre) [inline examples/memcached/cookbooks/custom-memcached/README.md]: the `perform_install` flag should be set to true

File: examples/memcached/cookbooks/custom-memcached/README.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  B1. Install on all app instances
+  B1.1. memcached should be running on app_master
+  B1.2. memcached should be running on app instances
+  B1.3. memcached should not be running on utility or database instances
+  B1.4. /data/app_name/shared/config/memcached.yml should list all the app instances in the environment
+  B2. Install on a utility instance named 'memcached'
+  B2.1. memcached should be running on the utility instance named memcached
+  B2.2. memcached should not be running on app_master, app, and database instances
+  B2.3. /data/app_name/shared/config/memcached.yml should list all the utility instances named memcached in the environment
+C. Install from package
```
Comment (dvalfre) [inline examples/memcached/cookbooks/custom-memcached/README.md]: the `perform_install` should be set to `true`

File: examples/memcached/cookbooks/custom-memcached/README.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+         # application
+
+    git clone https://github.com/engineyard/ey-cookbooks-stable-v5
+    cd ey-cookbooks-stable-v5
+    cp examples/memcached/cookbooks/custom-memcached /path/to/app/cookbooks/
+```
+
+If you do not have `cookbooks/ey-custom` on your app repository, you can copy
+`examples/memcached/cookbooks/ey-custom` to `/path/to/app/cookbooks` as well.
+
```
Comment (dvalfre) [inline examples/memcached/cookbooks/custom-memcached/README.md]: At some point the user should switch `perform_install` to `true`.  It is not necessarily a customization, so not belonging to that section of the Readme.

**Comment by radamanthus** [comment]
I updated stuff, squashed the commits, and did a `push --force`. Here's a summary of the changes:

1) Removed the dead link in memcached/README.md

2) Updated the Installation section in examples/memcached/cookbooks/custom-memcached/README.md based on the feedback

3) examples/memcached/cookbooks/custom-memcached/attributes/default.rb - perform_install is set to true. If the customer is adding custom-memcached, we assume he plans to install it :)

**Pull request merged**